containers/ws/release.sh

```diff
@@ -8,24 +8,52 @@ if [ -z "${RUNC:-}" ]; then
     }
 fi
 
-$RUNC build -t quay.io/cockpit/ws:release containers/ws
+# only build on host architecture per default
+if [ -z "${ARCHS:-}" ]; then
+    case $(uname -m) in
+        x86_64) ARCHS="amd64" ;;
+        arm64) ARCHS="arm64" ;;
+    esac
+fi
+for arch in $ARCHS; do
+    $RUNC buildx build --platform="linux/${arch}" -t "quay.io/cockpit/ws:release-${arch}" containers/ws
+done
 
 # smoke test
-name=ws-release
-$RUNC run --name $name -p 19999:9090 -d quay.io/cockpit/ws:release
-until curl --fail --show-error -k --head https://localhost:19999; do
-    sleep 1
+first_arch=$(echo "${ARCHS}" | cut -d " " -f 1)
+test_container=ws-release
+$RUNC run --name "${test_container}" -p 19999:9090 -d "quay.io/cockpit/ws:release-${first_arch}"
+
+for i in $(seq 1 30); do
+    curl --fail --show-error -k --head https://localhost:19999 && break
+
+    [ "${i}" -eq 30 ] && $RUNC rm -f "${test_container}" && exit 1 || sleep 1
```

Comment:
This chaining of && and || is rather brittle, it duplicates the `rm` command and the retry count. If this times out, then I deliberately did *not* remove the container, as that makes interactive debugging a lot simpler.  And we don't have to clean it up on a GitHub runner.

I would just keep the previous indefinite loop and let the workflow timeout take care of it, but if you want to make this explicit, please either consider counting down (`seq 30 -1 0`) or set a `success=1` flag or so.